modules/consensus/accept_test.go

```diff
@@ -27,60 +36,248 @@ func (db mockDbTx) Bucket(name []byte) dbBucket {
+	for _, pu := range m.p {
+		if bytes.Equal(b[:], pu.serialized[:]) {
+			pv, ok := v.(*processedBlock)
+			if !ok {
+				panic("mockBlockMarshaler.Unmarshal expected v to be of type processedBlock")
+			}
+			*pv = pu.unmarshaled
+			return pu.err
+		}
+	}
+	panic("unmarshal failed: predefined unmarshal not found")
+}
+
+// AddPredefinedUnmarshal adds a predefinedBlockUnmarshal to mockBlockMarshaler.
+func (m *mockBlockMarshaler) AddPredefinedUnmarshal(u predefinedBlockUnmarshal) {
+	m.p = append(m.p, u)
+}
+
+// mockBlockRuleHelper is an implementation of the blockRuleHelper interface for
+// unit testing.
+type mockBlockRuleHelper struct {
+	minTimestamp types.Timestamp
+}
+
+// minimumValidChildTimestamp returns the minimum timestamp of pb that can be
+// considered a valid block.
+func (brh mockBlockRuleHelper) minimumValidChildTimestamp(blockMap dbBucket, pb *processedBlock) types.Timestamp {
+	return brh.minTimestamp
+}
+
+// mockBlockValidator is an implementation of the blockValidator interface for
+// unit testing.
+type mockBlockValidator struct {
+	err error
+}
+
+// validateBlockParams stores the set of parameters passed to ValidateBlock
+type validateBlockParams struct {
+	called       bool
+	b            types.Block
+	minTimestamp types.Timestamp
+	target       types.Target
+	height       types.BlockHeight
+}
+
+// validateBlockParamsGot stores the parameters passed to the most recent call
+// to mockBlockValidator.ValidateBlock.
+var validateBlockParamsGot validateBlockParams
+
+// ValidateBlock stores the parameters it receives and returns the mock error
+// defined by mockBlockValidator.err.
+func (bv mockBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight) error {
+	validateBlockParamsGot = validateBlockParams{true, b, minTimestamp, target, height}
+	return bv.err
+}
+
+// mockParentID returns a mock BlockID value.
+func mockParentID() (parentID types.BlockID) {
+	parentID[0] = 42
+	return parentID
+}
+
+// mockParent returns a mock processedBlock with its ChildTarget member
+// initialized to a dummy value.
+func mockParent() (parent processedBlock) {
+	var mockTarget types.Target
+	mockTarget[0] = 56
+	parent.ChildTarget = mockTarget
+	return parent
+}
+
 var (
-	// validBlock is a mock valid Block.
-	validBlock = types.Block{Timestamp: 100}
+	validBlock = types.Block{
+		Timestamp: 100,
+		ParentID:  mockParentID(),
+	}
+	invalidBlock = types.Block{
+		Timestamp: 500,
+		ParentID:  mockParentID(),
+	}
+	// parentBlockSerialized is a mock serialized form of a processedBlock.
+	parentBlockSerialized = []byte{3, 2, 1}
 	// emptyDosBlocks is a dosBlocks map where no block is marked bad.
-	emptyDosBlocks = map[types.BlockID]struct{}{}
-	// emptyTx is a transaction associated with an empty block map.
-	emptyTx = mockDbTx{map[string]dbBucket{
-		string(BlockMap): mockDbBucket{},
-	}}
+	emptyDosBlocks         = map[types.BlockID]struct{}{}
```

Comment:
I don't think this variable is necessary, would much rather see it spelled out in the structs.

also, you should be using `make` when creating maps.

`= make(map[types.BlockID]struct{})`